File: Mathematical Algorithms/Exponential Squaring (Fast Modulo Multiplication)/ExpSq.cpp
#define N 1000000007L; // prime modulo value

long exponentiation(long base, long exp) {
    long t = 1L;
    while (exp > 0) {

        // for cases where exponent
        // is not an even value
        if (exp % 2 != 0)
            t = (t * base) % N;

        base = (base * base) % N;
        exp /= 2;
    }
    return t % N;
}

File: Mathematical Algorithms/Exponential Squaring (Fast Modulo Multiplication)/ExpSq.py
N = 1000000007


def exponentiation(bas, exp):
    t = 1
    while exp > 0:

        # for cases where exponent
        # is not an even value
        if exp % 2 != 0:
            t = (t * bas) % N

        bas = (bas * bas) % N
        exp = int(exp / 2)
    return t % N